src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -455,6 +430,43 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
             return ReadFunctions(metadatas, descriptorProviders);
         }
 
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+        {
+            if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
+            }
+            else
+            {
+                // First see if there is an explicit primary file indicated
+                // in config. If so use that.
+                string functionPrimary = null;
+                string scriptFileName = (string)functionConfig["scriptFile"];
```

Comment:
This property used to be called "source". We're renaming this to "sourceFile". It's an undocumented property at this point, so we'll be safe in making the breaking change.